src/llmcompressor/modifiers/awq/base.py

```diff
@@ -96,53 +112,138 @@ class AWQModifier(Modifier):
+
+        model._symmetric = next(iter(symmetric_set))
+
+        group_size_set = set(
+            group.weights.group_size
+            for group in config_groups.values()
+            if group.weights is not None
+        )
+        assert (
+            len(group_size_set) == 1
+        ), "In AWQ, all config groups must use the same configuration for group_size"
+
+        model._group_size = next(iter(group_size_set))
+
+        return model
 
     def on_initialize(self, state: State, **kwargs) -> bool:
         """
-        Initialize and run AWQ on the given state
+        Initialize AWQ on the given state
 
         :param state: state to run AWQ on
         :return: True on a successful run, False otherwise
         """
 
+        # apply config to model and prepare calibration hooks
+        if QuantizationMixin.has_config(self):
+            QuantizationMixin.initialize_quantization(self, state.model)
+
         self._set_resolved_mappings(state.model)
 
         with calibration_forward_context(state.model):
             self._set_module_kwargs(state.model, state.data.calib)
 
-        self._setup_scale_hooks()
-        with calibration_forward_context(state.model):
-            self._calibrate(state.model, state.data.calib)
-        self.remove_hooks()
-        self._concat_collected_activations()
+        return True
 
-        with calibration_forward_context(state.model):
-            self._apply_smoothing(state.model)
+    def on_start(self, state: State, event: Event, **kwargs):
+        self.started_ = True
 
-        return True
+        # register quantization calibration hooks
+        # assume quantization has been initialized by this modifier or one before it
+        QuantizationMixin.start_calibration(self, state.model)
+        # Unlike qmod, do not quantize as we calibrate
+        # This choice does not seem to have a meaningful impact on accuracy
+        state.model.apply(disable_quantization)
+        self._setup_activation_cache_hooks()
+
+    def on_event(self, state: State, event: Event, **kwargs):
+        if event.type_ == EventType.CALIBRATION_EPOCH_START:
+            if not self.started_:
+                self.on_start(state, None)
+
+        elif event.type_ == EventType.SEQUENTIAL_EPOCH_START:
+            self._activations.clear()
+
+        elif event.type_ == EventType.SEQUENTIAL_EPOCH_END:
+            self._concat_collected_activations()
+
+            with calibration_forward_context(state.model), HooksMixin.disable_hooks():
+                self._apply_smoothing(state.model)
+
+            self._activations.clear()
+
+        elif event.type_ == EventType.CALIBRATION_EPOCH_END:
+            if not self.ended_:
+                self.on_end(state, None)
+
+    def on_end(self, state: State, event: Event, **kwargs):
+        """
+        Finish calibrating by removing observers and calibration hooks
+        """
+        self.ended_ = True
+
```

Comment:
I typically add a warning here to check if there are any `_activations` that haven't been used to scale yet

Reply:
added